docs/design/2023-02-23-cloud-native-architecture.md

```diff
@@ -0,0 +1,148 @@
+- Author: [flowbehappy](http://github.com/flowbehappy)
+
+## Solve what problem?
+
+The current TiFlash architecture is a typical shared nothing architecture, which brings some drawbacks:
+
+1. Storage and computation cannot be scaled separately
+2. The update operation and the read operation are loaded on the same nodes and affect each other, including IO, CPU and other systems resources. The task of AP query is usually heavy and has a high suddenness.
+3. Slow down scaling. Scaling needs to do data synchronization between TiFlash nodes.
+4. The cost-efficiency is not optimal. For example, there are only queries at night, but we need to keep TiFlash instances around.
+
+
+## Basic ideas
+![](images/2023-02-23-cloud-native-architecture-1.png)
+
+We split TiFlash node into the following nodes:
+
+1. <b>Write Node (WN)</b> is responsible for storing raft logs, decoding the data, and writing it to memtable. The data written to disk by memtable will be uploaded to S3 for persistence. And it triggers remote compaction as appropriate. It also provides data shopshot information for query, such as data files involved in query.
+2. <b>Read Node (RN)</b> is responsible for the query. To start the query, it need to first request Writes Nodes to obtain the necessary snapshot information and Memtable data in memory, and then directly request S3 to run queries, no need to interact with Write Nodes any more. It uses the local disk that comes with the ec2 node to do data caching to improve query performance.
+3. <b>Compaction Worker</b> is used to handle various background tasks from Write Nodes offload to avoid affecting the foreground thread. Since it is a stateless node, it is very convenient to apply for resources dynamically and avoid compaction delays.
+
+
+
+## Advantages
+1. Storage and calculation separation reduces deployment costs
+    * Storage and computing resources can be scaled separately on demand to avoid resource mismatch
+    * Light and heavy queries can be supported by different Query Tenant respectively. For example, regular heavy calculations can be temporarily started to complete the calculation and destroyed after use.
+2. Read and write separation improves stability. Synchronize data, organize logic, and query load are physical separated, which can avoid heavy AP query load crowding out system resources.
+3. Greatly increase the speed of volume expansion and contraction
+    * Because RN is stateless, in theory, expansion and Load Balance can be completed in a few seconds.
+    * The Region migration on WN does not need to resynchronize the snapshot from the TiKV , and the migration is completed directly by downloading or referencing the S3 file. It is expected that a new node can complete the Load Balance within 1min.
+4. Improves Scalability while retaining the performance benefits of Locality
+    * RN can use the local disk as LRU cache, so that most of the hot data does not need to read S3
```

Comment:
```suggestion
4. Improves Scalability while retaining the performance benefits of locality
    * RN can use the local disk as an LRU cache, so that most of the hot data does not need to read S3
```